src/client/client.cpp

```diff
@@ -350,17 +359,15 @@ void Client::Loop( const InputState& input_state, const bool paused )
 			camera_controller_.GetAcceleration( input_state.keyboard, move_direction, move_acceleration );
 
 			Messages::PlayerMove message;
-			message.view_direction= AngleToMessageAngle( camera_controller_.GetViewAngleZ() + Constants::half_pi );
-			message.move_direction= AngleToMessageAngle( move_direction );
-			message.acceleration= static_cast<unsigned char>( move_acceleration * 254.5f );
-			message.jump_pressed= camera_controller_.JumpPressed();
-			message.weapon_index= requested_weapon_index_;
-
-			message.view_dir_angle_x= AngleToMessageAngle( camera_controller_.GetViewAngleX() );
-			message.view_dir_angle_z= AngleToMessageAngle( camera_controller_.GetViewAngleZ() );
-			message.shoot_pressed= shoot_pressed_;
-			message.color= settings_.GetOrSetInt( SettingsKeys::player_color );
-
+			message.view_direction   = AngleToMessageAngle( camera_controller_.GetViewAngleZ() + Constants::half_pi );
+			message.move_direction   = AngleToMessageAngle( move_direction );
+			message.acceleration     = static_cast<unsigned char>( move_acceleration * 254.5f );
+			message.jump_pressed     = input_state.mouse[ static_cast<unsigned int>( SystemEvent::MouseKeyEvent::Button::Right ) ] || camera_controller_.JumpPressed();
```

Comment:
Are you sure we should support jumping via mouse right button? Maybe it will be better to assign some else to this button? For example, weapon zoom, (bit for now there is no such zoom).